src/python/WMCore/WMBS/MySQL/Jobs/LoadForErrorHandler.py

```diff
@@ -40,12 +40,84 @@ class LoadForErrorHandler(DBFormatter):
                      INNER JOIN wmbs_file_details parent ON parent.id = wfp.parent
                      WHERE wfp.child = :fileid """
 
+    jobMask = """SELECT DISTINCT FirstEvent, LastEvent, FirstLumi, LastLumi, FirstRun,
+                                LastRun FROM wmbs_job_mask WHERE job = :jobid"""
+
     runLumiSQL = """SELECT fileid, run, lumi, num_events FROM wmbs_file_runlumi_map
                      WHERE fileid = :fileid"""
 
+    runLumiSQLWithMask = """SELECT fileid, run, lumi, num_events FROM wmbs_file_runlumi_map
+                         WHERE fileid = :fileid AND run = :run AND 
+                         lumi >= :firstLumi and lumi <= :lastLumi"""
+
+    def getJobMask(self, jobID, conn=None, transaction=False):
+
+        maskResult = self.dbi.processData(self.jobMask, jobID, conn=conn, transaction=transaction)
+        return self.formatDict(maskResult)
+
+    def getMaskedRunLumis(self, fileID, mask, conn=None, transaction=False):
+        # create run lumi bind:
+
+        binds = []
+        for m in mask:
+            if m["FirstLumi"] > -1 and m["LastLumi"] > -1 and m["FirstRun"] > -1 and m["LastRun"] > -1:
+                # FirstRun and LastRun should be the same but just in case
+                binds.extend([{"fileid": fileID, "firstLumi": m["FirstLumi"], "firstLumi": m["FirstLumi"], "run": run}
+                              for run in range(m["FirstRun"], m["LastRun"] + 1)])
+        if binds:
+            runLumiResult = self.dbi.processData(self.runLumiSQLWithMask, binds, conn=conn, transaction=transaction)
+        return runLumiResult
 
-    def getRunLumis(self, fileBinds, fileList,
-                    conn=None, transaction=False):
+    def getNoMaskRunLumis(self, fileID, conn=None, transaction=False):
+        """
+        _getNoRunLumis_
+
+        Fetch run/lumi/events information for each file and append Run objects
+        to the files information.
+        """
+        lumiResult = self.dbi.processData(self.runLumiSQL, {'fileid': fileID}, conn=conn,
+                                          transaction=transaction)
+        return self.formatDict(lumiResult)
+
+    def formatRunLumi(self, runLumiResult):
+
+        fileRuns = {}
+        for l in runLumiResult:
+            run = l['run']
+            lumi = l['lumi']
+            numEvents = l['num_events']
+            fileRuns.setdefault(run, [])
+            fileRuns[run].append((lumi, numEvents))
+
+        runLumiList = []
+        for r in fileRuns.keys():
+            newRun = Run(runNumber=r)
+            newRun.lumis = fileRuns[r]
+            runLumiList.append(newRun)
+        return runLumiList
+
+    def updateFilesWithRunLumi(self, jobID, fileObj, conn=None, transaction=False):
+
+        mask = self.getJobMask(jobID, conn=None, transaction=False)
+        if mask:
+            #asssumes ther is only one first/last events mask information per job and file.
+            if mask[0]['LastEvent'] and mask[0]['FirstEvent']:
```

Comment:
one comment: the lines 104-116 show 3 (!!!) different notation styles, UpperCase, key_attribute, lowerUpperCase.
Could you please choose one and use it across the code and make it consistent with code style (which I think is lowerUpperCase).

Reply:
Unfortunately this is from schema definition.
https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/WMBS/CreateWMBSBase.py#L328

Worse yet, Oracle ignores the upper case (convert to lower case) but MariaDB is not. I could change whole code wise but maybe not this time. (This need to be patched current agent)